<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PasswordGeneratorWidget</class>
 <widget class="QDialog" name="PasswordGeneratorWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>404</width>
    <height>496</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Password Generator</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Method:</string>
       </property>
       <property name="buddy">
        <cstring>cmbMethod</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cmbMethod">
       <item>
        <property name="text">
         <string>Characters</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Templates</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btnMethodInfo">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="resources.qrc">
         <normaloff>:/help.png</normaloff>:/help.png</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QWidget" name="charactersWidget" native="true">
     <layout class="QGridLayout" name="gridLayout">
      <item row="2" column="0">
       <widget class="QCheckBox" name="chkCharaz">
        <property name="text">
         <string>a-z</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QCheckBox" name="chkCharNum">
        <property name="text">
         <string>0-9</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="chkCharAZ">
        <property name="text">
         <string>A-Z</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QLineEdit" name="txtCharCustom"/>
      </item>
      <item row="5" column="0">
       <widget class="QCheckBox" name="chkCharSpecial">
        <property name="text">
         <string>@&amp;&amp;%?,=:+*$#!'^~;/.</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Select Characters:</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Custom Characters:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>txtCharCustom</cstring>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QCheckBox" name="chkCharBracket">
        <property name="text">
         <string>()[]{}&lt;&gt;</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QCheckBox" name="chkCharUnderline">
        <property name="text">
         <string>_ (Underline)</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QCheckBox" name="chkCharSpace">
        <property name="text">
         <string>(Space)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QCheckBox" name="chkCharMinus">
        <property name="text">
         <string>- (Minus)</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QSpinBox" name="txtPasswordLength">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>12</number>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Password Length:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>txtPasswordLength</cstring>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="templatesWidget" native="true">
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="2" column="0">
       <spacer name="verticalSpacer_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Password Length:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>txtPasswordLengthTemplate</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="txtPasswordLengthTemplate">
        <property name="minimum">
         <number>4</number>
        </property>
        <property name="maximum">
         <number>40</number>
        </property>
        <property name="value">
         <number>14</number>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QCheckBox" name="chkTemplateSpecial">
        <property name="text">
         <string>Add special Character</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="chkTemplateNumber">
        <property name="text">
         <string>Add Number</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Password Strength</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="lblEntropy">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblStrength">
        <property name="text">
         <string/>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="btnGenerate">
       <property name="text">
        <string>Generate</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="txtPassword">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Close</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>cmbMethod</tabstop>
  <tabstop>btnMethodInfo</tabstop>
  <tabstop>chkCharaz</tabstop>
  <tabstop>chkCharAZ</tabstop>
  <tabstop>chkCharNum</tabstop>
  <tabstop>chkCharSpecial</tabstop>
  <tabstop>chkCharUnderline</tabstop>
  <tabstop>chkCharMinus</tabstop>
  <tabstop>chkCharSpace</tabstop>
  <tabstop>chkCharBracket</tabstop>
  <tabstop>txtCharCustom</tabstop>
  <tabstop>txtPasswordLength</tabstop>
  <tabstop>chkTemplateSpecial</tabstop>
  <tabstop>chkTemplateNumber</tabstop>
  <tabstop>txtPasswordLengthTemplate</tabstop>
  <tabstop>btnGenerate</tabstop>
  <tabstop>txtPassword</tabstop>
 </tabstops>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>cmbMethod</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>147</x>
     <y>31</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>11</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkCharaz</sender>
   <signal>toggled(bool)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>54</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>2</x>
     <y>98</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkCharAZ</sender>
   <signal>toggled(bool)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>35</x>
     <y>121</y>
    </hint>
    <hint type="destinationlabel">
     <x>2</x>
     <y>125</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkCharNum</sender>
   <signal>toggled(bool)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>41</x>
     <y>144</y>
    </hint>
    <hint type="destinationlabel">
     <x>1</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkCharSpecial</sender>
   <signal>toggled(bool)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>168</y>
    </hint>
    <hint type="destinationlabel">
     <x>6</x>
     <y>183</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkCharUnderline</sender>
   <signal>toggled(bool)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkCharMinus</sender>
   <signal>toggled(bool)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>294</x>
     <y>121</y>
    </hint>
    <hint type="destinationlabel">
     <x>403</x>
     <y>133</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkCharSpace</sender>
   <signal>toggled(bool)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>361</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>403</x>
     <y>162</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkCharBracket</sender>
   <signal>toggled(bool)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>288</x>
     <y>168</y>
    </hint>
    <hint type="destinationlabel">
     <x>403</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtCharCustom</sender>
   <signal>textChanged(QString)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>215</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtPasswordLength</sender>
   <signal>valueChanged(int)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>361</x>
     <y>235</y>
    </hint>
    <hint type="destinationlabel">
     <x>403</x>
     <y>247</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnGenerate</sender>
   <signal>clicked()</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>generateClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>37</x>
     <y>462</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>154</x>
     <y>496</y>
    </hint>
    <hint type="destinationlabel">
     <x>134</x>
     <y>405</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>274</x>
     <y>496</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>401</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnMethodInfo</sender>
   <signal>clicked()</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>showMethodExplanation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>207</x>
     <y>29</y>
    </hint>
    <hint type="destinationlabel">
     <x>336</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtPasswordLengthTemplate</sender>
   <signal>valueChanged(int)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>385</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkTemplateSpecial</sender>
   <signal>toggled(bool)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>126</x>
     <y>283</y>
    </hint>
    <hint type="destinationlabel">
     <x>402</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkTemplateNumber</sender>
   <signal>toggled(bool)</signal>
   <receiver>PasswordGeneratorWidget</receiver>
   <slot>uiChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>309</x>
     <y>286</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>uiChanged()</slot>
  <slot>generateClicked()</slot>
  <slot>showMethodExplanation()</slot>
 </slots>
</ui>
